src/main/java/openblocks/common/tileentity/TileEntityTank.java

```diff
@@ -329,6 +329,8 @@ private void tryBalanceNeighbors(FluidStack contents) {
 			sum += n.tank.getFluidAmount();
 
 		final int suggestedAmount = sum / (count + 1);
+		if (Math.abs(suggestedAmount - contents.amount) < 5) return; // Don't balance small amounts to reduce server load
+
```

Comment:
I'll split them.

This is an older change I did early on my server after doing some profiling (I don't have the data anymore though). That fixed lags caused by some players using those tanks as medium-load buffers (3x3x3). Those continued ticking for long periods of time, depending on the situation.
After this change the tanks worked a lot better for buffer-use.

You're right with the draining-problem, you will have some leftovers. But during my testing that was also true without my change (of course only 1mb per tank instead of up to 5mb). But at least for my player base that was an acceptable tradeoff for the reduced block-lag ;)